charts/karpenter/templates/role.yaml

```diff
@@ -14,19 +14,7 @@ rules:
   - apiGroups: ["coordination.k8s.io"]
     resources: ["leases"]
     verbs: ["get", "watch"]
-{{- if .Values.webhook.enabled }}
```

Comment:
@charliedmcb do we need to retain this permission for the MW case?

Reply:
If we arent toggling our _attempt_ to read the MW off (I dont think we are?) then we'll get an auth error if we dont have the perms.

Reply:
We will attempt to read the MW ConfigMap in all cases. However, its gated on one condition beforehand, which is the assumption the `SYSTEM_NAMESPACE` is set in the Env:
- [Reading SYSTEM_NAMESPACE](https://github.com/Azure/karpenter-provider-azure/blob/84e449787ec72268efb0c7af81ec87a6b3ee95fa/pkg/controllers/nodeclass/status/images.go#L68)
- [Gate on it being set](https://github.com/Azure/karpenter-provider-azure/blob/84e449787ec72268efb0c7af81ec87a6b3ee95fa/pkg/controllers/nodeclass/status/images.go#L181)

So, we wouldn't necessarily fail, unless if the `SYSTEM_NAMESPACE` is set. However, I agree keeping the perms would be best, regardless.

Reply:
Added configmap read permission